lib/client/user/user-client.js

```diff
@@ -12,28 +12,55 @@ var userClient;
 var UserClient = function(mediator) {
   this.mediator = mediator;
   this.initComplete = false;
+  this.readyPromise = this.ready();
   this.initPromise = this.init();
 
   //Flag to cache user verification.
   this.userVerified = false;
 };
 
-var xhr = function(_options) {
+/**
+ *
+ * Checking for the $fh SDK to be ready before any requests can be made.
+ *
+ * @returns {*}
+ */
+UserClient.prototype.ready = function() {
+  var deferred = q.defer();
+  var self = this;
+
+  $fh.on('fhinit', function(error) {
+    if (error) {
+      deferred.reject(new Error(error));
+      return;
+    }
+    self.appid = $fh.getFHParams().appid;
+    self.initComplete = true;
+    deferred.resolve();
+  });
+
+  return deferred.promise;
+};
```

Comment:
No new deferreds please :cry: 

We can use `q.Promise((resolve, reject) => {})` to have code more similar to A+, making for an easier migration.